### ISSUE TITLE
Incorrect advice for changing the code when `Style/CollectionQuerying` occurs

### ISSUE THREAD

**Issue by Yegorov** [rubocop/rubocop#15067]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/CollectionQuerying

### Steps to reproduce the problem

example.rb
```ruby
# frozen_string_literal: true

x = [false]
x.count.zero?
x.count.positive?
```

### Expected behavior

Using [`none?`](https://docs.ruby-lang.org/en/4.0/Array.html#method-i-none-3F) and [`any?`](https://docs.ruby-lang.org/en/4.0/Array.html#method-i-any-3F) in all cases is incorrect, because:
```ruby
x = [false]
x.count.zero? => false
x.none? => true
x.empty? => false

x = [false]
x.count.positive? => true
x.any? => false
!x.empty? => true
```
This is due to how these methods process the contents of the array (see docs).
I suggest fixing `none?` to `empty?` and `any?` to `!empty?`.

### Actual behavior

```
$ rubocop ./example.rb --debug
For /home/artem/Desktop/rubocop: configuration from /home/artem/Desktop/rubocop/.rubocop.yml
Default configuration from /home/artem/.local/share/mise/installs/ruby/4.0.1/lib/ruby/gems/4.0.0/gems/rubocop-1.86.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /home/artem/Desktop/rubocop/example.rb
Loading cache from /home/artem/.cache/rubocop_cache/965651ce79abd3832c43b3b0d80d6b778385b773/6d7a3b621ca1730e04accd938619e4bdab66cfb1/ea347b025419094b4be8d264a86f33a5f784de80
C

Offenses:

example.rb:4:3: C: [Correctable] Style/CollectionQuerying: Use none? instead.
x.count.zero?
  ^^^^^^^^^^^
example.rb:5:3: C: [Correctable] Style/CollectionQuerying: Use any? instead.
x.count.positive?
  ^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
Finished in 0.0664 seconds
```

### RuboCop configuration

```yaml
AllCops:
  NewCops: enable
```

### RuboCop version

```shell
1.86.0 (using Parser 3.3.11.1, Prism 1.9.0, rubocop-ast 1.49.1, analyzing as Ruby 4.0, running on ruby 4.0.1) [x86_64-linux]
```
